### DISCUSSION TITLE
Channel ID in Voice Leave event

### DISCUSSION THREAD

**Discussion by kvizyx** [disgoorg/disgo#357] [Help] [answered]
I want to keep track of members count of specific voice channel in my app without using session id. How can i achieve that? The problem is that in Voice Leave event data "channel id" is always nil. Doesn't Discord provide id of channel from which user left?

**Comment by topi314** [comment, marked as answer]
Discord always sends the current channel a user would join
`nil` means they left. 

To see how many users are in a voice channel you'd either need to track all voice states which can be done by adding 
```go
client, err := disgo.New(...,
 bot.WithCacheConfigOpts(
  cache.WithCaches(cache.FlagVoiceStates)
 )
)
```


Then you can just access all members in a voice channel via
```go
client.Caches().AudioChannelMembers(channel)
```

**Reply by kvizyx** [reply]
Thanks!
